src/main/java/com/faforever/client/game/CreateGameController.java

```diff
@@ -377,7 +393,8 @@ public void onRandomMapButtonClicked() {
 
   public void onGenerateMapButtonClicked() {
     try {
-      mapGeneratorService.generateMap().thenAccept(mapName -> {
+      byte spawnCount = (byte) preferencesService.getPreferences().getGeneratorPrefs().getSpawnCountProperty();
```

Comment:
In theory this always equals the currently selected value, but it feels a bit wrong. Maybe read this value from the spawnCountSpinner instead?

Reply:
Well I could. Makes no difference...